### PR TITLE
(MAINT) bump ruby version

### PR DESCRIPTION
**Pull request by chelnak**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,74 +1,520 @@
...
-    EnforcedStyle: same_line
-Layout/MultilineHashBraceLayout:
-    EnforcedStyle: new_line
-Layout/MultilineMethodDefinitionBraceLayout:
-    EnforcedStyle: new_line
-
-Style/AutoResourceCleanup:
-    Enabled: true
+  DisplayCopNames: true
+  TargetRubyVersion: '2.5'
```
Comment (bastelfreak) [inline .rubocop.yml:7]: Given that Ruby 2.5 is EoL, should we drop support for that as well and. go strait to 2.6 or 2.7?

Comment (chelnak) [reply .rubocop.yml:7]: I'm very tempted. I really don't like supporting EOL things.

Let me discuss internally tomorrow to collect some more views.

Comment (david22swan) [reply .rubocop.yml:7]: Go for it

File: Gemfile [hunk 2]
```diff
@@ -7,39 +7,19 @@ group :test do
...
-      gem 'rack', '~> 1.0', :require => false
-    end
-
-    gem 'pry'
-  end
+    gem 'github_changelog_generator', require: false
+    gem 'faraday-retry', require: false
+    gem 'rubocop', '~> 1.6.1', require: false
+    gem 'rubocop-rspec', '~> 2.0.1', require: false
+    gem 'rubocop-performance', '~> 1.9.1', require: false
```
Comment (ekohl) [inline Gemfile:23]: In some places I've created a rubocop group.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ jobs:
         with:
           ruby-version: 2.7
           bundler-cache: true
-      - run: bundle exec rake rubocop
+      - run: bundle exec rubocop
```
Comment (ekohl) [inline .github/workflows/ci.yml:21]:
```suggestion
      - run: bundle exec rubocop
```
```suggestion
      - run: bundle exec rubocop --format github
```

File: Rakefile [hunk 4]
```diff
@@ -1,46 +1,19 @@
-require 'rake'
-require 'rspec/core/rake_task'
-require 'puppet-lint'
-require 'puppet-lint/tasks/release_test'
 require 'bundler/gem_tasks'
+require 'rubocop/rake_task'
+require 'github_changelog_generator/task'
```
Comment (ekohl) [inline Rakefile:3]: I'd keep this optional and implement fallback. This allows you to skip the development group in CI.

File: puppet-lint.gemspec [hunk 5]
```diff
@@ -1,20 +1,36 @@
...
+    'Puppet, Inc.',
+    'Community Contributors',
+  ]
+  spec.email = [
+    'tim@sharpe.id.au',
+    'modules-team@puppet.com',
+  ]
+  spec.license = 'MIT'
+
+  spec.required_ruby_version = Gem::Requirement.new(">= 2.5".freeze)
```
Comment (ekohl) [inline puppet-lint.gemspec:35]: Rubocop is 2.7+. Those should match.

File: puppet-lint.gemspec [hunk 6]
```diff
@@ -1,20 +1,36 @@
...
-  s.executables = s.files.grep(%r{\Abin/}) { |f| File.basename(f) }
-  s.require_paths = ['lib']
+  spec.files = Dir[
+    'README.md',
+    'LICENSE',
+    'lib/**/*',
+    'bin/**/*',
+    'spec/**/*',
+  ]
+  spec.test_files = Dir['spec/**/*']
```
Comment (ekohl) [inline puppet-lint.gemspec:20]: `test_files` is ignored nowadays and recent versions of Rubocop suggest to remove this.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (chelnak) [reply .github/workflows/ci.yml]: This workflow is about to get changed.. however I love the suggestion of `--format github`.

I'll implement it in the newer workflows.

Comment (ekohl) [reply .github/workflows/ci.yml]: I just realized I didn't link to https://docs.rubocop.org/rubocop/1.22/formatters.html#github-actions-formatter, which I intended.

File: Rakefile (hunk 4 above)
Comment (chelnak) [reply Rakefile:3]: Do you have an example for this suggestion?

File: Gemfile (hunk 2 above)
Comment (chelnak) [reply Gemfile:23]: I like this - will add now.

File: puppet-lint.gemspec (hunk 6 above)
Comment (chelnak) [reply puppet-lint.gemspec:20]: 👍

File: puppet-lint.gemspec (hunk 5 above)
Comment (chelnak) [reply puppet-lint.gemspec:35]: 👍

File: Rakefile (hunk 4 above)
Comment (ekohl) [reply Rakefile:3]: What I typically do is this:

```ruby
begin
  require 'github_changelog_generator/task'
rescue LoadError
  # Gem not present
else
  GitHubChangelogGenerator::RakeTask.new(:changelog)
end
```

The benefit of putting the definition in the else part is that you very narrowly scope the exception catching.

Comment (chelnak) [reply Rakefile:3]: Appreciate this thanks.

**Comment by chelnak** [comment]
FYI - I'm fully expecting the tests to be a sea of red here.

This PR is one step in a series of clean up PRs.

**Review by david22swan — APPROVED** [review]
LGTM

**Pull request merged**